Mutagen.Bethesda.Analyzers.Skyrim/Record/ConditionAnalyzer.cs

```diff
@@ -0,0 +1,82 @@
+﻿using Mutagen.Bethesda.Analyzers.SDK.Analyzers;
+using Mutagen.Bethesda.Analyzers.SDK.Results;
+using Mutagen.Bethesda.Analyzers.SDK.Topics;
+using Mutagen.Bethesda.Skyrim;
+namespace Mutagen.Bethesda.Analyzers.Skyrim.Record;
+
+public class ConditionAnalyzer : IIsolatedRecordAnalyzer<ISkyrimMajorRecordGetter>
+{
+    private static readonly TopicDefinition<string?> InvalidConditionReference = MutagenTopicBuilder.DevelopmentTopic(
+            "Condition Runs on Null Reference",
+            Severity.Error)
+        .WithFormatting<string?>("Condition {0} runs on reference, but reference is null");
+
+    public IEnumerable<TopicDefinition> Topics { get; } = [InvalidConditionReference];
+
+    public RecordAnalyzerResult? AnalyzeRecord(IsolatedRecordAnalyzerParams<ISkyrimMajorRecordGetter> param)
+    {
+        var conditions = param.Record.GetConditions();
+        if (conditions is null) return null;
+
+        var result = new RecordAnalyzerResult();
+        foreach (var condition in conditions)
+        {
+            if (condition.Data.RunOnType != Condition.RunOnType.Reference) continue;
+            if (!condition.Data.Reference.IsNull) continue;
+
+            switch (condition.Data)
+            {
+                case IGetEventDataConditionDataGetter getEventData:
+                    result.AddTopic(
+                        RecordTopic.Create(
+                            param.Record,
+                            InvalidConditionReference.Format(getEventData.Function.ToString()),
+                            x => x
+                        )
+                    );
+                    break;
+                case {} conditionData:
+                    result.AddTopic(
+                        RecordTopic.Create(
+                            param.Record,
+                            InvalidConditionReference.Format(conditionData.Function.ToString()),
+                            x => x
+                        )
+                    );
+                    break;
+            }
+        }
+
+        return result;
+    }
+}
+
+public static class RecordExtension
+{
+    public static IEnumerable<IConditionGetter>? GetConditions(this ISkyrimMajorRecordGetter record)
+    {
+        return record switch
+        {
+            ICameraPathGetter cameraPath => cameraPath.Conditions,
+            IConstructibleObjectGetter constructibleObject => constructibleObject.Conditions,
+            IDialogResponsesGetter dialogResponses => dialogResponses.Conditions,
+            IFactionGetter faction => faction.Conditions,
+            IIdleAnimationGetter idleAnimation => idleAnimation.Conditions,
+            ILoadScreenGetter loadScreen => loadScreen.Conditions,
+            IMagicEffectGetter magicEffect => magicEffect.Conditions,
```

Comment:
I would like to find a more automatically maintaining system for ones like this.  As games are being developed (Starfield for, ex), these lists will be hard to keep updated to the changing definitions

If this gets too bad, we can either:
- Add a `GetConditions` to mutagen
- Maybe even make a source generator.  Something like `GetFieldsOfType<IConditionGetter>(rec)`, where the source generator would analyze the fields of the record type and make a `GetConditions` style function on the fly.